### PR TITLE
Dedupe error source paths

### PR DESCRIPTION
**Pull request by rcebulko**
Band-aid until https://github.com/ampproject/amphtml/pull/27665 repairs the broken source-maps. Re-writes source path URLs to remove duplicated paths.

### CODE REVIEW

File: utils/stacktrace/frame.js [hunk 1]
```diff
@@ -24,11 +24,28 @@ class Frame {
...
+   * Temporary fix until #27665 fixes the sourcemaps.
+   * @param {string} source
+   * @return {string}
+   */
+  dedupeSource(source) {
+    if (!source) {
+      return source;
+    }
+
+    const matches = source.match(/ampproject\/amphtml\/\d+\/(.+)\/\1\/[^\/]+$/);
```
Comment (jridgewell) [inline utils/stacktrace/frame.js:43]: We can combine both the match and replace, which'll make it faster:

```js
source = source.replace(/(ampproject\/amphtml\/\d+)(\/.+)\2(\/[^\/]+)$/, '$1$2$3')
```

File: utils/stacktrace/frame.js [hunk 2]
```diff
@@ -24,11 +24,28 @@ class Frame {
    */
   constructor(name, source, line, column) {
     this.name = name;
-    this.source = source;
+    this.source = this.dedupeSource(source);
```
Comment (jridgewell) [inline utils/stacktrace/frame.js:27]: Nit: should be a static method or just a module function.

File: test/unit/test-frame.js [hunk 3]
```diff
@@ -32,4 +32,28 @@ describe('Frame', () => {
...
+    it('fixes duplicated source paths', () => {
+      const samples = [{
+        duped: 'https://raw.githubusercontent.com/ampproject/amphtml/2004142326360/extensions/amp-viewer-integration/0.1/messaging/extensions/amp-viewer-integration/0.1/messaging/messaging.js',
+        fixed: 'https://raw.githubusercontent.com/ampproject/amphtml/2004142326360/extensions/amp-viewer-integration/0.1/messaging/messaging.js',
+      }, {
+        duped: 'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/service/src/service/storage-impl.js',
+        fixed: 'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/service/storage-impl.js',
+      }, {
+        duped: 'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/src/chunk.js',
+        fixed: 'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/chunk.js',
```
Comment (jridgewell) [inline test/unit/test-frame.js:46]: Please add a test case that does not contain a duplicate

File: utils/stacktrace/frame.js (hunk 1 above)
Comment (rcebulko) [reply utils/stacktrace/frame.js:43]: Done.

File: utils/stacktrace/frame.js (hunk 2 above)
Comment (rcebulko) [reply utils/stacktrace/frame.js:27]: Doen.

File: test/unit/test-frame.js (hunk 3 above)
Comment (rcebulko) [reply test/unit/test-frame.js]: Done.

File: test/unit/test-frame.js [hunk 4]
```diff
@@ -32,4 +32,45 @@ describe('Frame', () => {
...
+        },
+        {
+          duped: 'unexpected-token.js',
+          fixed: 'unexpected-token.js',
+        },
+      ];
+
+      samples.forEach(({ duped, fixed }) => {
+        const frame = new Frame('name', duped, '1', '2');
+        console.log(duped, fixed);
```
Comment (jridgewell) [inline test/unit/test-frame.js:71]: Nit: if you're trying to log individual test cases, you could just procedurally generate the tests:

```js
samples.forEach({ duped, fixed }, index) => {
  it(`fixes duplicated source paths: ${index}`, () => {
    // do test here.
  });
});
```

Comment (rcebulko) [reply test/unit/test-frame.js:71]: In Jest there's an `it.each` method for this purpose, but I didn't see a mocha equivalent 😕. I'll make this change and submit.
